stdlib.sh

```diff
@@ -61,6 +61,113 @@ puts_verbose() {
   fi
 }
 
+# Key Value Store Utility
+# -------------------
+
+# This is a utility module that allows you to create a file that serves
+# as a key-value store that can be persisted to disk.
+#
+# It works around using lines formatted like:
+# 
+# key=value
+#
+# and using grep when searching for keys to find the last entry. This 
+# allows an existing key to be overwritten.
+#
+# This is inspired by the "world's simplest database" in Ch 3 of
+# Designing Data-Intensive Applications https://dataintensive.net/
+
+# Ensure that the file that we would like to persist this store to
+# exists. Also handles creating the containing directory if it doesn't
+# exist yet. 
+#
+# Will not overwrite the store if it already exists
+# Usage: kv_create file
+kv_create() {
+  local f=$1
+  mkdir -p "$(dirname "$f")"
+  touch "$f"
+}
+
+# Delete all keys and values from the store
+# Usage: kv_clear file
+kv_clear() {
+  local f=$1
+  echo "" > "$f"
+}
+
+# Set value for a key
+# Usage: kv_set file key value
+kv_set() {
+  if [[ $# -eq 3 ]]; then
+    local f=$1
+    if [[ -f $f ]]; then
+      echo "$2=$3" >> "$f"
```

Comment:
this breaks with newlines in `$3`, so I think `printf "$2=%q\n" "$3"` is better here; that stores it as escaped

Reply:
I went down this path, but couldn't make everything work the way I wanted. In the end I decided to flatten any multiline strings.

Most of my hesitation here comes from the likely need to parse these files in go or ruby soon and wanting to keep the escaping rules very simple